### PR TITLE
feature(makefile): Makefiles cross-compile arm

### LINKED ISSUE #2050: use local go env to build sealer binary
## Issue Description
image

Type: *feature request*

### PR DESCRIPTION
**Pull request by cubxxw**
### Describe what this PR does / why we need it

Makefiles cross-compile arm

### Does this pull request fix one issue?

Fixes #2050 use local go env to build sealer binary

### Describe how to verify it
```bash
❯ ls ./_output/bin/sealer/linux_arm64/sealer -al
-rwxr-xr-x 1 root root 52035584 Mar 28 14:05 ./_output/bin/sealer/linux_arm64/sealer
```

### Special notes for reviews
no

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,13 +25,13 @@ build: clean ## build binaries by default
 	@echo "build sealer and seautil bin"
 	hack/build.sh
 
-linux: clean ## build binaries for linux
-	@echo "build sealer and seautil bin for linux"
-	GOOS=linux GOARCH=amd64 hack/build.sh $(GitTag)
+linux: clean ## Build binaries for Linux
+	@echo "Building sealer and seautil binaries for Linux (amd64)"
+	GOOS=linux GOARCH=amd64 hack/build.sh $(GIT_TAG)
```
Comment (kakaZhou719) [inline Makefile:30]: could we use local go env `GOARCH` to replace it ? make it as a variable

Comment (Stevent-fei) [reply Makefile]: `GIT_TAG` is nil?

Comment (cubxxw) [reply Makefile]: I didn't change it, and I wonder why, but it didn't affect my make.I incorporate it into later tasks

File: Makefile [hunk 2]
```diff
@@ -23,15 +28,19 @@ style: fmt vet lint ## code style: fmt,vet,lint
...
-	@echo "build sealer and seautil bin for linux"
-	GOOS=linux GOARCH=amd64 hack/build.sh $(GitTag)
+linux-amd: clean ## Build binaries for Linux (amd64)
+	@echo "Building sealer and seautil binaries for Linux (amd64)"
+	GOOS=linux GOARCH=amd64 $(TOOLS_DIR) $(GIT_TAG)
 
-linux-arm64: clean ## build binaries for linux
-	@echo "build sealer and seautil bin for linux"
-	GOOS=linux GOARCH=arm64 hack/build.sh $(GitTag)
+linux-arm: clean ## Build binaries for Linux (arm64)
```
[the remark this answers is not in the document]
Comment (cubxxw) [reply Makefile:41]: Of course, but the help here will not retrieve the following number after adding the book and output the help information normally. If CICD needs to be used, I can change it. If not, I hope to change this point in the later design

File: Makefile (hunk 1 above)
Comment (Stevent-fei) [reply Makefile]: maybe this variable is deleted, you can optimize it

Comment (cubxxw) [reply Makefile]: yep

File: Makefile [hunk 3]
```diff
@@ -111,3 +134,12 @@ deepcopy:install-deepcopy-gen
 	  -O zz_generated.deepcopy   \
 	  --go-header-file "$(HEAD_FILE)" \
 	  --output-base "${GOPATH}/src"
+
+## help: Display help information
+help: Makefile
```
Comment (kakaZhou719) [inline Makefile:139]: nice

File: Makefile [hunk 4]
```diff
@@ -7,44 +14,58 @@ else
...
+	@echo "===========> run e2e test for sealer bin"
+	scripts/test-sealer.sh
 
-clean: ## clean
-	@rm -rf _output
+## clean: Remove all files that are created by building. 
+.PHONY: clean
+clean:
+	@echo "===========> Cleaning all build output"
+	@-rm -rf _output
```
Comment (kakaZhou719) [inline Makefile:65]: @rm -rf _output

Comment (cubxxw) [reply Makefile:60]: make clean use of '-' to avoid errors where the directory does not exist

**Review by kakaZhou719 — APPROVED** [review]
LGTM

**Pull request merged**